gattc_sd.go

```diff
@@ -137,19 +157,19 @@ func (s *DeviceService) DiscoverCharacteristics(uuids []UUID) ([]DeviceCharacter
 		return nil, errAlreadyDiscovering
 	}
 
-	// Make a list of UUIDs in SoftDevice short form, for easier comparing.
-	shortUUIDs := make([]C.ble_uuid_t, len(uuids))
-	for i, uuid := range uuids {
+	// Make a map of UUIDs in SoftDevice short form, for easier comparing.
+	shortUUIDs := make(map[uuidWrapper]C.ble_uuid_t)
```

Comment:
Why this change to use a map instead? Maps are expensive, especially in TinyGo (they use a lot of heap space and are super hard to optimize). Also, it looks like the key is not used making the hashmap useless.

Reply:
This was to simplify the code to search if was already in the list, but could be replaced with a function to just look thru the entire slice.

Reply:
Changed.